docs/APIs/communication/network_sockets.md

```diff
@@ -63,6 +63,6 @@ typedef enum nsapi_error {
 
 The network-socket API also supports nonblocking operations. The ``set_blocking`` member function changes the state of a socket. When a socket is in nonblocking mode, socket operations return ``NSAPI_ERROR_WOULD_BLOCK`` when a transaction cannot be immediately completed.
 
-To allow efficient use of nonblocking operations, the socket classes provide an ``attach`` member function to register a callback on socket state changes. The callback will be called when the socket can successfully receive, send or accept, or when an error occurs. The callback may be called spuriously without reason.
+To allow efficient use of nonblocking operations, the socket classes provide an ``attach`` member function to register a callback on socket state changes. When the socket can successfully receive, send or accept, or when an error occurs, the system triggers a callback. It may call the callback spuriously without reason.
 
-The callback may be called in an interrupt context and should not perform expensive operations such as receiving and sending calls.
+The callback may be called in interrupt context and should not perform operations such as receiving and sending calls. Instead, take precautions accordingly and do not make any read or write calls until it is on a thread.
```

Comment:
Not wild about a generic "take precautions". Is there anything other than waiting until the callback is on a thread? If not, then just talk about waiting for the thread. If there is anything else, we should probably explain it or reference some other explanation.